homeassistant/components/homekit_controller/__init__.py

```diff
@@ -224,7 +224,12 @@ def setup(self):
                 if service['iid'] != self._iid:
                     continue
                 for char in service['characteristics']:
-                    uuid = CharacteristicsTypes.get_uuid(char['type'])
+                    try:
+                        uuid = CharacteristicsTypes.get_uuid(char['type'])
+                    except KeyError:
```

Comment:
Should we handle it here or in the library?

Reply:
Good question! I feel like this is multiple questions in one too!

Should the library return unknown characteristics? Probably. If you worked out how to use the characteristic you would be able to drive it with the API's we have.

Should the library normalize the characteristics itself? Yeah, probably. I think it doesn't just because these fragments can be pulled directly from the JSON the device itself is returning, so no one has ever thought to normalize things, they are just passed on as is. I've put that ony my list to discuss with the maintainer.

Should CharacteristicsTypes raise a KeyError in this case? I think so, though we can get rid of it if the API tries to normalize uuid for us.

I think the hotfix should land as-is in the meantime, it will take a while to sort this out upstream if maintainer agrees.